### ISSUE TITLE
Loss nan while training (custom)

### ISSUE THREAD

**Issue by dmaugis** [michhar/ssd-pytorch-custom#2]
using torch 0.4.1.post2, training on a small custom dataset Loss goes to nan after a few iterations.
What did I do wrong please ?

> python train.py --dataset Custom --dataset_root data/image_data/train --cuda True --save_folder weights --num_workers 0 --start_iter 0 --batch_size 1
/ssd
Loading base network...
Initializing weights...
train.py:239: UserWarning: nn.init.xavier_uniform is now deprecated in favor of nn.init.xavier_uniform_.
  init.xavier_uniform(param)
Loading the dataset...
Training SSD on: CUSTOM
Using the specified args:
Namespace(basenet='vgg16_reducedfc.pth', batch_size=1, cuda=True, dataset='Custom', dataset_root='data/image_data/train', gamma=0.1, lr=0.001, momentum=0.9, num_workers=0, resume=None, save_folder='weights', start_iter=0, visdom=False, weight_decay=0.0005)
Data loader length... 55
/root/.pyenv/versions/3.7.0/lib/python3.7/site-packages/torch/nn/functional.py:52: UserWarning: size_average and reduce args will be deprecated, please use reduction='sum' instead.
  warnings.warn(warning.format(ret))
timer: 0.0397 sec.
iter 0 || Loss: 20.1227 || timer: 0.0328 sec.
iter 2 || Loss: 15.4248 || timer: 0.0305 sec.
iter 4 || Loss: 16.2171 || timer: 0.0331 sec.
iter 6 || Loss: 57.7096 || timer: 0.0311 sec.
iter 8 || Loss: 27.9619 || timer: 0.0317 sec.
iter 10 || Loss: 82.4413 || timer: 0.0308 sec.
iter 12 || Loss: 14.3774 || timer: 0.0332 sec.
iter 14 || Loss: nan || timer: 0.0315 sec.
iter 16 || Loss: nan || timer: 0.0342 sec.
<

**Comment by michhar** [comment]
@dmaugis Thanks for checking out the project.  I'm updating in the next few weeks and trying to resolve this very issue.  I will post back here when it is updated.
